File: Tugas Pratikum/Python/Perpustakaan.py
class Node:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None
class Node:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None

class Perpustakaan:
    def __init__(self):
        self.root = None

    def insert(self, root, value):
        if root is None:
            root = Node(value)
            return root
        
        if str(value) <= str(root.value):
            root.left = self.insert(root.left, value)
        else:
            root.right = self.insert(root.right, value)

        return root

    def preorderTraversal(self, node):
        if node is not None:
            print(node.value)
            self.preorderTraversal(node.left)
            self.preorderTraversal(node.right)

    def inorderTraversal(self, node):
        if node is not None:
            self.inorderTraversal(node.left)
            print(node.value)
            self.inorderTraversal(node.right)

    def postorderTraversal(self, node):
        if node is not None:
            self.postorderTraversal(node.left)
            self.postorderTraversal(node.right)
            print(node.value)

binaryTree = Perpustakaan()
while True:
    print("=== BOOK ORGANIZING SYSTEM ===")
    print("1. Insert Book Data")
    print("2. Show Book Data")
    print("3. Exit")
    menu = int(input("Pilih menu: "))
    if menu == 1:
        print("\n-------------------------------------------")
        print("Rules: \n- Gunakan underscore sebagai pengganti spasi. Misal: Bumi_Manusia\n- Inputan bisa berupa judul buku atau ID buku")
        n = int(input("Jumlah data buku yang akan dimasukkan: "))
        for i in range(n):
            value = input("Data buku ke-" + str(i + 1) + ": ")
            binaryTree.root = binaryTree.insert(binaryTree.root, value)
    elif menu == 2:
        print("\n-------------------------------------------")
        print("Susunan Buku dengan Preorder Traversal:")
        binaryTree.preorderTraversal(binaryTree.root)
        print("\nSusunan Buku dengan Inorder Traversal:")
        binaryTree.inorderTraversal(binaryTree.root)
        print("\nSusunan Buku dengan Postorder Traversal:")
        binaryTree.postorderTraversal(binaryTree.root)
    elif menu == 3:
        break
    else:
        print("Invalid Menu")

File: Tugas Pratikum/Java/BinaryTreeScheme/src/Perpustakaan.java
import java.util.Scanner;

public class Perpustakaan<T> {
    Node<T> root;

    Node insert(Node<T> root, T value) {
        if (root == null) {
            root = new Node(value);
            return root;
        }
        if (value.toString().compareTo(root.value.toString()) <= 0) {
            root.left = insert(root.left, value);
        } else {
            root.right = insert(root.right, value);
        }

        return root;
    }

    void preorderTraversal(Node<T> node) {
        if (node != null) {
            System.out.println(node.value);
            preorderTraversal(node.left);
            preorderTraversal(node.right);
        }
    }

    void inorderTraversal(Node<T> node) {
        if (node != null) {
            inorderTraversal(node.left);
            System.out.println(node.value);
            inorderTraversal(node.right);
        }
    }

    void postorderTraversal(Node<T> node) {
        if (node != null) {
            postorderTraversal(node.left);
            postorderTraversal(node.right);
            System.out.println(node.value);
        }
    }

    public static void main(String[] args) {
        Perpustakaan binaryTree = new Perpustakaan();
        Scanner scanner = new Scanner(System.in);
        while (true) {
            System.out.println("=== BOOK ORGANIZING SYSTEM ===");
            System.out.println("1. Insert Book Data");
            System.out.println("2. Show Book Data");
            System.out.println("3. Exit");
            System.out.print("Pilih menu: ");
            int menu = scanner.nextInt();
            if (menu == 1) {
                System.out.println("\n-------------------------------------------");
                System.out.println("Rules: \n- Gunakan underscore sebagai pengganti spasi. Misal: Bumi_Manusia\n- Inputan bisa berupa judul buku atau ID buku");
                System.out.print("Jumlah data buku yang akan dimasukkan: ");
                int n = scanner.nextInt();
                for (int i = 0; i < n; i++) {
                    System.out.print("Data buku ke-" + (i + 1) + ": ");
                    String value = scanner.next();
                    binaryTree.root = binaryTree.insert(binaryTree.root, value);
                }
            } else if (menu == 2) {
                System.out.println("\n-------------------------------------------");
                System.out.println("Susunan Buku dengan Preorder Traversal:");
                binaryTree.preorderTraversal(binaryTree.root);
                System.out.println("\nSusunan Buku dengan Inorder Traversal:");
                binaryTree.inorderTraversal(binaryTree.root);
                System.out.println("\nSusunan Buku dengan Postorder Traversal:");
                binaryTree.postorderTraversal(binaryTree.root);
            } else if (menu == 3) {
                break;
            } else {
                System.out.println("Invalid Menu");
            }
        }
    }
}